kornia/geometry/epipolar/_metrics.py

```diff
@@ -12,45 +12,47 @@ def sampson_epipolar_distance(
 
     Args:
         pts1: correspondences from the left images with shape
-          (B, N, 2 or 3). If they are not homogeneous, converted automatically.
+          (*, N, 2 or 3). If they are not homogeneous, converted automatically.
         pts2: correspondences from the right images with shape
-          (B, N, 2 or 3). If they are not homogeneous, converted automatically.
-        Fm: Fundamental matrices with shape :math:`(B, 3, 3)`. Called Fm to
+          (*, N, 2 or 3). If they are not homogeneous, converted automatically.
+        Fm: Fundamental matrices with shape :math:`(*, 3, 3)`. Called Fm to
           avoid ambiguity with torch.nn.functional.
         squared: if True (default), the squared distance is returned.
         eps: Small constant for safe sqrt.
 
     Returns:
-        the computed Sampson distance with shape :math:`(B, N)`.
+        the computed Sampson distance with shape :math:`(*, N)`.
 
     """
     if not isinstance(Fm, torch.Tensor):
         raise TypeError(f"Fm type is not a torch.Tensor. Got {type(Fm)}")
 
-    if (len(Fm.shape) != 3) or not Fm.shape[-2:] == (3, 3):
+    if (len(Fm.shape) < 3) or not Fm.shape[-2:] == (3, 3):
         raise ValueError(f"Fm must be a (*, 3, 3) tensor. Got {Fm.shape}")
 
-    if pts1.size(-1) == 2:
+    if pts1.shape[-1] == 2:
         pts1 = convert_points_to_homogeneous(pts1)
 
-    if pts2.size(-1) == 2:
+    if pts2.shape[-1] == 2:
         pts2 = convert_points_to_homogeneous(pts2)
 
     # From Hartley and Zisserman, Sampson error (11.9)
     # sam =  (x'^T F x) ** 2 / (  (((Fx)_1**2) + (Fx)_2**2)) +  (((F^Tx')_1**2) + (F^Tx')_2**2)) )
 
-    # line1_in_2: torch.Tensor = (F @ pts1.permute(0,2,1)).permute(0,2,1)
-    # line2_in_1: torch.Tensor = (F.permute(0,2,1) @ pts2.permute(0,2,1)).permute(0,2,1)
+    # line1_in_2: torch.Tensor = (F @ pts1.transpose(dim0=-2, dim1=-1)).transpose(dim0=-2, dim1=-1)
+    # line2_in_1: torch.Tensor = (F.transpose(dim0=-2, dim1=-1) @ pts2.transpose(dim0=-2, dim1=-1)).transpose(
+    #     dim0=-2, dim1=-1
+    # )
     # Instead we can just transpose F once and switch the order of multiplication
-    F_t: torch.Tensor = Fm.permute(0, 2, 1)
+    F_t: torch.Tensor = Fm.transpose(dim0=-2, dim1=-1)
     line1_in_2: torch.Tensor = pts1 @ F_t
     line2_in_1: torch.Tensor = pts2 @ Fm
 
     # numerator = (x'^T F x) ** 2
-    numerator: torch.Tensor = (pts2 * line1_in_2).sum(2).pow(2)
+    numerator: torch.Tensor = (pts2 * line1_in_2).sum(dim=-1).pow(2)
```

Comment:
```suggestion
    numerator: torch.Tensor = (pts2 * line1_in_2).sum(-1).pow(2)
```
I'd suggest to not use kwargs to make it more pythonic and not torch specific. We are exploring how to refine so that we can reuse codes for other python backends like numpy, jax or mxnet ;)